### ISSUE TITLE
Provide single-usage container apps

### ISSUE THREAD

**Issue by QuaiGoner** [games-on-whales/gow#100]
Hello. Hope its not too rude to post it here, but it would be much easier and cleaner for new users to just add to their docker-compose files another container like RetroArch, which contains xorg,pulse,udev,sunshine and app in following form:

```
 gow-retroarch
    image:  gow-retroarch:latest
    container_name:  gow-retroarch101
    group_add:
      - 109 #OR whatever required for video and graphics
    environment:
      - TZ=Europe/London
      - SUNSHINE_USER=admin
      - SUNSHINE_PASS=admin
      - SUNSHINE_WEB_UI=49999
      - SUNSHINE_ENCODER=vaapi
      - XORG_RESOLUTION: 1280x720
      - XORG_CURRENT_OUTPUT: HDMI-0 #OR some other virtual output
      - XORG_REFRESH_RATE: 60
      - XORG_FORCE_RESOLUTION: true
    volumes:
      - ./config/gow-retroarch101/config:/config #Root folder with subfolders like /retroarch, /sunshine...
      - /mnt/Games:/Games:ro
      - /dev/input:/dev/input:ro
      - udev:/run/udev/:ro
    devices:
      # VAAPI/GFX Devices
      - /dev/dri/renderD128:/dev/dri/renderD128
      - /dev/dri/card0:/dev/dri/card0
      - /dev/fb0:/dev/fb0
    network_mode: host
    privileged: true
#all other ports required by sunshine
    restart: unless-stopped
```

**Comment by ABeltramo** [comment]
TLDR: This has been requested multiple times, unfortunately it's the result of a series of limitations that we are trying to overcome in the upcoming [games-on-whales/wolf](https://github.com/games-on-whales/wolf).

## The issue

On a basic level, you can bundle Xorg, Sunshine and any GUI app into a single container. If this is what you really are after I suggest you to look into [Steam-Headless/docker-steam-headless](https://github.com/Steam-Headless/docker-steam-headless) which provides a single Docker container with everything needed in a single bundle (note: doesn't use Sunshine, but it could be added).

There are several issues of going with this approach:
 - You can't easily compose multiple applications. 
   - AFAIK Xorg can't be running multiple times. This means that if, for example, you want to run both a Firefox image and a Retroarch image you'll have to stop one container and start the other one.
   - Since one container bundles everything, stopping it will also close the remote streaming session.
   - How many applications should we "bundle" together into a single docker image? This looks more like a VM than a Docker image.
 - You can't use the host desktop environment and just run Steam or Sunshine via Docker like you can right now in GOW.
 - You have no way to support multiple independent users/streams.
   - This is also a problem deeply tied to Sunshine, which brings us to the next point: 

## The solution

We would like to solve a very narrow and specific use case: allow multiple clients to connect and share a single headless server for graphical (HW accelerated) apps.  

In order to achieve that, I started working on [Wolf](https://github.com/games-on-whales/wolf); unfortunately, it's still under heavy development and far from being easily usable like Sunshine is.  
It currently supports streaming video and audio via Moonlight, and it's multi-user already. We are now trying to figure out how to [go fully headless](https://github.com/games-on-whales/wolf/issues/7) and support multiple independent screens for different users.  

What I would really like to have is a single `wolf` container that will then spin up additional containers (like `steam`, `retroarch`, etc.) and all the necessary (virtual) display/audio when a user starts the relative app directly from Moonlight.

This would probably solve your feature request and make everything "easier".  
We are working on it 🚧 , stay tuned!

**Comment by ABeltramo** [comment]
Checkout [Wolf](https://github.com/games-on-whales/wolf)!
